src/app/context.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import React, { ReactNode, createContext, useContext, useState } from 'react';
+import { NewsWithId } from './models/types';
+
+type NewsState = {
+  context: NewsWithId[] | null;
+  setContext(news: NewsWithId[]): void;
+};
+
+const initialNewsState = {
+  context: null,
```

Comment:
Why do you use `null` as a default value? Would not it be better to use `[]`?

Reply:
Null - search wasn't invoked, [ ] - search was invoked, but nothing was found.